### ISSUE TITLE
Delete hyper-parameters from web-app

### ISSUE THREAD

**Issue by nirraviv** [allegroai/clearml#258]
I'm currently using `trains==0.16.2` and registering hyper-parameters from a dictionary using `task.connect(config)`.

After a preliminary run from the terminal I'm trying to remove one hyper-parameter by explicitly deleting the argument from Configuration -> General and by deleting its content (that should set it as None?) but unfortunately in both the parameter is as the original run.

Any idea how to solve this problem?

**Comment by jkhenning** [comment]
Hi @nirraviv ,

> unfortunately in both the parameter is as the original run

Do you mean that after deleting it in the UI it immediately appears back in the UI?

**Comment by nirraviv** [comment]
@jkhenning, if I'm deleting from the UI it appears back, and if I just set as empty it stays empty in the UI but the value when printing is not None (but the original value from the dictionary).

**Comment by bmartinn** [comment]
Hi @nirraviv 
Just making sure I understand the issue,
You have an experiment in the system (created by running your code manually)
You then clone the experiment in the UI, and change a parameter value to empty (i.e. None in most cases), then you send the experiment to be executed by `trains-agent` but the parameter stays the same?
Is that correct ?

Background:
When creating an experiment from code (i.e. running the python code manually) all the configurations are stored in `trains-server` , this is a one way street, whatever you do in the UI will be overridden if the code is reusing (overwriting) the experiment. When `trains-agent` executes the code, the opposites happens and the parameters from the `trains-server` (UI) override the parameters in code.

**Comment by nirraviv** [comment]
Hi @bmartinn ,
You are correct. 
I attached simple example.

Steps to reproduce the problems:
**Experiment 1**: 
- Run the given code and clone a new task in the Web-APP.
- *Delete* (trash symbol) `temp` from Configuration -> General and enqueue this task.
- The `temp=123` variable will appear again and will be printed in the Log tab (`tmp: 123`).

**Experiment 2**:
- Same as in (1)
- *Empty* `temp`.
- The `temp` will stay empty in Configuration -> General but `tmp: 123` will be printed in the Log tab (even though I expect it to be None).

[trains_mnist.py.txt](https://github.com/allegroai/trains/files/5606262/trains_mnist.py.txt)

**Comment by bmartinn** [comment]
Thank you @nirraviv , the code snippet really helped!

Experiment (1): This is the expected behavior. when a field is missing the remote execution will use the default value and update back the UI.
Experiment (2): This is the actual bug, the empty value should be cast to None, as the code demonstrates it is not, and we end up with the default value.

**Comment by bmartinn** [comment]
Hi @nirraviv 
Could you maybe verify the fix solved the issue?
```
pip install git+https://github.com/allegroai/trains.git
```
Notice that if you are running remotely, make sure that in the "installed packages" section you have:
```
git+https://github.com/allegroai/trains.git
```

**Comment by nirraviv** [comment]
Hi @bmartinn, 
The current version indeed fix the problem.

Just to be sure, the trash icon in the configuration tab will only work if I added a new hyperparameter and then delete it?

**Comment by bmartinn** [comment]
Hi @nirraviv 
>  trash icon in the configuration tab will only work if I added a new hyperparameter and then delete it?

The trash icon will remove the parameter from the UI, however it will **not** remove it from the dictionary itself (or `argparse` for that matter). Basically if there is no parameter, the default value will be updated back. Although it is possible to support variable removal, we feel it will allow you to break the code way too easily.
WDYT?

p.s.

RC is now out with the fix
```
pip install trains==0.16.4rc0
```

**Comment by nirraviv** [comment]
Hi @bmartinn ,
I agree that removing variable may be dangerous and empty fields should be enough for most of the cases, but in this case you should consider removing the trash icon since it is very confusing.

**Comment by bmartinn** [comment]
>  you should consider removing the trash icon since it is very confusing.

Good point 👍 
It should probably be "remove icon", and maybe a tool tip explaining what will happen in remote execution.
Since you can add any parameter to a section, it makes sense to have the opposite action, remove a parameter you might have mistakenly added...

**Comment by allegroai-git** [comment]
Closing as this was released in Trains v0.16.4
